gdal/frmts/netcdf/netcdfsgwriterutil.cpp

```diff
@@ -196,8 +208,127 @@ namespace nccfdriver
+
+        node_coordinates_dimID = ncdf.nc_def_vdim(nodecoord_name.c_str(), 1);
+
+        // Node Coordinates - Variable Names
+        err_code = nc_get_att_text(ncID, containerVar_realID, CF_SG_NODE_COORDINATES, node_coord_names);
+        NCDF_ERR(err_code);
+        if (err_code != NC_NOERR)
+        {
+            throw SGWriter_Exception_NCInqFailure(containerVarName.c_str(), CF_SG_NODE_COORDINATES, "varName");
+        }
+
+        // Node Count
+        if(geo != POINT)
+        {
+            std::string nodecount_name = containerVarName + "_" + std::string(CF_SG_NODE_COUNT);
+            node_count_dimID = ncdf.nc_def_vdim(nodecount_name.c_str(), 1);
+            node_count_varID = ncdf.nc_def_vvar(nodecount_name.c_str(), NC_INT, 1, &node_count_dimID);
+        }
+
+        // Do the same for part node count, if it exists
+        char pnc_name[NC_MAX_CHAR + 1] = {0};
+        err_code = nc_get_att_text(ncID, containerVar_realID, CF_SG_PART_NODE_COUNT, pnc_name);
+
+
+        if(err_code == NC_NOERR)
+        {
+            pnc_dimID = ncdf.nc_def_vdim(pnc_name, 1);
+            pnc_varID = ncdf.nc_def_vvar(pnc_name, NC_INT, 1, &pnc_dimID);
+
+            char ir_name[NC_MAX_CHAR + 1] = {0};
+            nc_get_att_text(ncID, containerVar_realID, CF_SG_INTERIOR_RING, ir_name);
+
+            // For interior ring too (for POLYGON and MULTIPOLYGON); there's always an assumption
+            // that interior rings really do exist until the very end in which case it's known whether or not
+            // that assumption was true or false (if false, this (and PNC attribute for Polygons) will just be deleted)
+            if(this->writableType == POLYGON || this->writableType == MULTIPOLYGON)
+            {
+                intring_varID = ncdf.nc_def_vvar(ir_name, NC_INT, 1, &pnc_dimID);
+            }
+        }
+
+        // Node coordinates Var Definitions
+        int new_varID;
+        CPLStringList aosNcoord(CSLTokenizeString2(node_coord_names, " ", 0));
+
+        if(aosNcoord.size() < 2)
+            throw SGWriter_Exception();
+
+        // first it's X
+        new_varID = ncdf.nc_def_vvar(aosNcoord[0], NC_DOUBLE, 1, &node_coordinates_dimID);
+        ncdf.nc_put_vatt_text(new_varID, CF_AXIS, CF_SG_X_AXIS);
+
+        this->node_coordinates_varIDs.push_back(new_varID);
+
+        // second it's Y
+        new_varID = ncdf.nc_def_vvar(aosNcoord[1], NC_DOUBLE, 1, &node_coordinates_dimID);
+        ncdf.nc_put_vatt_text(new_varID, CF_AXIS, CF_SG_Y_AXIS);
+
+        this->node_coordinates_varIDs.push_back(new_varID);
+
+        // (and perhaps) third it's Z
+        if(aosNcoord.size() > 2)
+        {
+            new_varID = ncdf.nc_def_vvar(aosNcoord[2], NC_DOUBLE, 1, &node_coordinates_dimID);
+            ncdf.nc_put_vatt_text(new_varID, CF_AXIS, CF_SG_Z_AXIS);
+
+            this->node_coordinates_varIDs.push_back(new_varID);
+        }
+    }
+
+/*    void ncLayer_SG_Metadata::scanExistingContainer(int containerVID)
```

Comment:
to remove ?

Reply:
yeah, I had bigger plans for netCDFVID at first I guess...